qualtop/wizard/account_invoice_refund.py

```diff
@@ -0,0 +1,26 @@
+from odoo import models, api
+
+
+class AccountInvoiceRefund(models.TransientModel):
+    _inherit = "account.invoice.refund"
+
+    @api.multi
+    def invoice_refund(self):
+        res = super(AccountInvoiceRefund, self).invoice_refund()
+        data_refund = self.read(['filter_refund'])[0]['filter_refund']
+        if data_refund == 'refund' and res.get('domain', False):
+            """ To obtain the credit note id is necessary search in
+            res['domain'] which has the form
+            [('type', '=', 'out_refund'), ('id', 'in', [5])]. The credit note
+            id is in res['domain'][1][2][0]
+            """
+            cn_id = res['domain'][1][2][0]
+            credit_note = self.env['account.invoice'].search(
+                [('id', '=', cn_id)])
+            inv_id = dict(self._context or {}).get('active_id')
```

Comment:
This could lead to inv_id been a False value because of the get()
then the proper treatment afterwards is:

> if not inv_id:
>    return res

Reply:
For new api is not required to specify `dict(self._context or {})`
You can use directly `self.env.context.get('active_id')`

If you want examples just run:
`rgrep "context.*active_id" . --include=*.py`

Reply:
Changes executed